src/mococrw/csr.h

```diff
@@ -38,8 +38,30 @@ class CertificateSigningRequest
     explicit CertificateSigningRequest(const DistinguishedName &distinguishedName,
                                        const AsymmetricKeypair &key);
 
+    /**
+     * Construct a new CertificateSigningRequest for the given distinguished name.
+     * The given public key is integrated in the CSR whereas the given private key
+     * is used to sign the CertificateSigningRequest. The signature uses the provided
+     * digestFunction.
+     */
+    explicit CertificateSigningRequest(const DistinguishedName &distinguishedName,
+                                       const AsymmetricKeypair &key,
+                                       const openssl::DigestTypes digestFunction);
+
+    /**
+     * Return the PEM for this CSR as a string.
+     */
+    std::string toPEM() const;
+
+    /**
+     * Return the DER for this CSR as a string.
```

Comment:
Technically, it's not a string, since it's a `std::vector<uint8_t>`, but I guess we could still stay that's a binary string.